plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java

```diff
@@ -86,15 +90,19 @@ public void widgetSelected(SelectionEvent event) {
         WorkbenchUtil.openInBrowser(PlatformUI.getWorkbench(), event.text);
       }
     });
-
+    
+    sdkVersionLabel = new Label(parent, SWT.LEFT);
+    sdkVersionLabel.setText(Messages.getString("SdkVersion", "Unset"));
+    
     if (CloudSdkManager.isManagedSdkFeatureEnabled()) {
       useLocalSdk = new Button(parent, SWT.CHECK);
       useLocalSdk.setText(Messages.getString("UseLocalSdk")); //$NON-NLS-1$
       useLocalSdk.addSelectionListener(new SelectionAdapter() {
         @Override
-        public void widgetSelected(SelectionEvent e) {
+        public void widgetSelected(SelectionEvent event) {
           if (!useLocalSdk.getSelection()) {
             status = Status.OK_STATUS;
+            updateSelectedVersion();
```

Comment:
Do you think we could extract the code from `doCheckState()` and have `updateSelectedVersion()` work for both cases?  This special casing, along with the special casing in `doCheckState()` make this a bit hard to reason through.

Reply:
Maybe. My initial attempt to do this didn't quite work, but I may be able to improve it a bit.